src/dht_proxy_server.cpp

```diff
@@ -205,8 +215,14 @@ DhtProxyServer::getNodeInfo(const std::shared_ptr<restbed::Session>& session) co
                     if (id)
                         result["id"] = id.toString();
                     result["node_id"] = dht_->getNodeId().toString();
-                    result["ipv4"] = dht_->getNodesStats(AF_INET).toJson();
-                    result["ipv6"] = dht_->getNodesStats(AF_INET6).toJson();
+                    if (ipv4Stats_.good_nodes == 0 && ipv6Stats_.good_nodes == 0) {
```

Comment:
* Shouldn't be needed (run the update on startup if required)
* Doesn't run on startup but every time there is no good known node which is not the intent

Reply:
Not sure what you mean by "Doesn't run on startup but every time there is no good known node which is not the intent". Here I update the stats only if there is no good known node.

Reply:
According to the comment the intent was to update on startup but that's not what the code does and there seems to be no reason to update only if there is no good node.

Reply:
In fact. If I update on startup, the dhtrunner can haven't discovered any node for now. Here I just want to be connected to the proxy as soon as possible (after dht.cpp discover some nodes in the network). I don't know if I'm clear enough?

Reply:
Yes with your explanation actually the code makes sense, and that's a good reason to update if there is no good known node: the node might go through a disconnected state, on startup or later, and in that case we want to update the state ASAP. Update the comments to make it clear.

For a long-term solution we might have multiple dht state listeners and update the stats in case of connectivity change.

Reply:
Yeah the comment was unclear. I'll fix this tomorrow morning.

Reply:
Done